### PR TITLE
feat: tg mini apps support

### PR DESCRIPTION
**Pull request by ganchoradkov**
## Description
Fixed deep links when in telegram mini apps context
- base64 payload encoding when targetted to https://t.me universal links
- `formatUriFromPairing` util to regenerate pairing uri from pairing to be used as deeplink
- `parseUri` util to handle incoming pairing uris in base64 encoding
- updated `@walletconnect/modal` to latest

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Draft PR (breaking/non-breaking change which needs more work for having a proper functionality _[Mark this PR as ready to review only when completely ready]_)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested?
in telegram mini apps
https://walletconnect.slack.com/archives/C04DB2EAHE3/p1726817747215279
`2.16.3-rc-tma-1`

## Checklist

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

# Additional Information (Optional)

Please include any additional information that may be useful for the reviewer.

### CODE REVIEW

File: packages/utils/src/misc.ts [hunk 1]
```diff
@@ -369,12 +369,17 @@ export async function handleDeeplinkRedirect({
...
     const json = typeof wcDeepLink === "string" ? JSON.parse(wcDeepLink) : wcDeepLink;
     let deeplink = json?.href;
-
     if (typeof deeplink !== "string") return;
 
+    const payload = `requestId=${id}&sessionTopic=${topic}`;
     if (deeplink.endsWith("/")) deeplink = deeplink.slice(0, -1);
-
-    const link = `${deeplink}/wc?requestId=${id}&sessionTopic=${topic}`;
+    let link = `${deeplink}`;
```
Comment (Cali93) [inline packages/utils/src/misc.ts]: Curious why do we need to interpolate this ?
```suggestion
    let link = deeplink;
```

File: packages/utils/src/misc.ts [hunk 2]
```diff
@@ -369,12 +369,17 @@ export async function handleDeeplinkRedirect({
...
+    if (deeplink.startsWith("https://t.me")) {
+      const startApp = deeplink.includes("?") ? "&startapp=" : "?startapp=";
+      link = `${link}${startApp}${toBase64(payload, true)}`;
+    } else {
+      link = `${link}/wc?${payload}`;
+    }
```
Comment (Cali93) [inline packages/utils/src/misc.ts]: I'd suggest to use the URL constructor as @enesozturk suggested on the other PR

File: packages/utils/src/uri.ts [hunk 3]
```diff
@@ -17,6 +18,13 @@ export function parseRelayParams(params: any, delimiter = "-"): RelayerTypes.Pro
 }
 
 export function parseUri(str: string): EngineTypes.UriParameters {
+  if (!str.includes("wc:")) {
```
Comment (Cali93) [inline packages/utils/src/uri.ts:21]: Super nit and not sure if it suits this use case but assuming the `str` always starts with `wc://...` you could also use the URL constructor here and check if the protocol is `wc:`

**Review by Cali93 — APPROVED** [review]
LGTM left minor comments

File: packages/utils/src/misc.ts (hunk 1 above)
Comment (ganchoradkov) [reply packages/utils/src/misc.ts]: leftover from previous behaviour

File: packages/utils/src/misc.ts (hunk 2 above)
Comment (ganchoradkov) [reply packages/utils/src/misc.ts]: this url approach doesn't seem to work well with native links, maybe i'm doing something wrong but assigning `/wc?` path results in tripple dashes e.g. 
```
const url = new URL('trust://');
url.pathname = "wc";
-> trust:///wc?

Comment (ganchoradkov) [reply packages/utils/src/misc.ts]: will keep the behaviour but added tests to confirm it works as expected

**Pull request merged**